v-next/hardhat/test/internal/builtin-plugins/network-manager/request-handlers/e2e.ts

```diff
@@ -0,0 +1,271 @@
+            type: "http",
+            url: "http://localhost:8545",
+            chainId: 1,
+            gas: FIXED_GAS_LIMIT,
+            gasPrice: FIXED_GAS_PRICE,
+            from: "0x2a97a65d5673a2c61e95ce33cecadf24f654f96d",
+          },
+        },
+      },
+      // List of methods that the handlers will call; we mock the responses
+      {
+        eth_chainId: "0x1",
+      },
+    );
+
+    // Use the localhost network for these tests because the modifier is only
+    // applicable to HTTP networks. EDR networks do not require this modifier.
+    const connection = await hre.network.connect("localhost");
+
+    const res = await connection.provider.request({
+      method: "eth_sendTransaction",
+      params: [
+        {
+          to: "0x0000000000000000000000000000000000000012",
+        },
+      ],
+    });
+
+    assert.ok(Array.isArray(res), "res should be an array");
+
+    // gas
+    assert.equal(res[0].gas, numberToHexString(FIXED_GAS_LIMIT));
+    // gasPrice
+    assert.equal(res[0].gasPrice, numberToHexString(FIXED_GAS_PRICE));
+    // sender
+    assert.equal(res[0].from, "0x2a97a65d5673a2c61e95ce33cecadf24f654f96d");
+  });
+
+  it("should successfully executes all the handlers that set automatic values", async () => {
+    // should use the handlers in this order: ChainIdValidatorHandler, AutomaticGasHandler, AutomaticGasPriceHandler, AutomaticSenderHandler
+
+    const MOCKED_GAS_LIMIT = 21000;
+    const GAS_MULTIPLIER = 1.337;
+    const LATEST_BASE_FEE_IN_MOCKED_PROVIDER = 80;
+
+    const hre = await createMockedNetworkHre(
+      {
+        networks: {
+          localhost: {
+            type: "http",
+            url: "http://localhost:8545",
+            chainId: 1,
+            gas: "auto",
+            gasMultiplier: GAS_MULTIPLIER,
+          },
+        },
+      },
+      {
+        // List of methods that the handlers will call; we mock the responses
+        eth_chainId: "0x1",
+        eth_getBlockByNumber: {
+          baseFeePerGas: "0x1",
+          gasLimit: numberToHexString(MOCKED_GAS_LIMIT * 1000),
+        },
+        eth_estimateGas: numberToHexString(MOCKED_GAS_LIMIT),
+        eth_feeHistory: {
+          baseFeePerGas: [
+            numberToHexString(LATEST_BASE_FEE_IN_MOCKED_PROVIDER),
+            numberToHexString(
+              Math.floor((LATEST_BASE_FEE_IN_MOCKED_PROVIDER * 9) / 8),
+            ),
+          ],
+          reward: [["0x4"]],
+        },
+        eth_accounts: ["0x123006d4548a3ac17d72b372ae1e416bf65b8eaf"],
+      },
+    );
+
+    // Use the localhost network for these tests because the modifier is only
+    // applicable to HTTP networks. EDR networks do not require this modifier.
+    const connection = await hre.network.connect("localhost");
+
+    const res = await connection.provider.request({
+      method: "eth_sendTransaction",
+      params: [
+        {
+          to: "0x0000000000000000000000000000000000000012",
+          maxFeePerGas: "0x99",
```

Comment:
this value here confuses me a bit. Why is it necessary?

Reply:
It's not strictly necessary. It's more of an additional check to be sure that there is no conflict with the logic if `maxFeePergas` is present. But it can be removed, I did it [here](https://github.com/NomicFoundation/hardhat/pull/5994/commits/71d8935de87bebcd7b06c7d6ec70a25b517dc415)